core/src/test/java/org/apache/calcite/test/ReflectiveSchemaTest.java

```diff
@@ -972,4 +980,48 @@ public static class DateColumnSchema {
     assertNotNull(statistic);
     assertThat(statistic.getRowCount(), is(2D));
   }
+
+  public static final ReflectiveSchema COLLECTIONS_SCHEMA =
+      new ReflectiveSchema(new PetCollectionsSchema());
+
+  public static Stream<Arguments> collectionsSchemaFieldNames() {
+    return Stream.of(
+        Arguments.of("cats_array_list"),
+        Arguments.of("cats_linked_list"),
+        Arguments.of("cats_stack"),
+        Arguments.of("cats_vector"),
+        Arguments.of("cats_array_dequeue"),
+        Arguments.of("cats_hash_set"),
+        Arguments.of("cats_linked_hash_set"),
+        Arguments.of("cats_tree_set"));
+  }
+
+  /**
+   * Test that the Collections works as field of ReflectiveSchema.
```

Comment:
"Test that Collections can be used as field"

Reply:
> "Test that Collections can be used as field"
Fixed